### PR TITLE
Link quay push secret to the SA in build-templates-e2e namespace

### PR DESCRIPTION
**Pull request by mkosiarc**

### CODE REVIEW

**Review by mmorhun — APPROVED** [review]

File: components/build-templates/base/e2e/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: appstudio-pipeline
+secrets:
+  - name: quay-push-secret
+imagePullSecrets:
+  - name: quay-push-secret
```
Comment (chmeliik) [inline components/build-templates/base/e2e/serviceaccount.yaml:8]: Does this cause any issues in the dev environment? quay-push-secret is only in stage and prod https://github.com/redhat-appstudio/infra-deployments/pull/2800

**Comment by dheerajodha** [comment]
The latest job failed on [RHTAPBUGS-985](https://issues.redhat.com/browse/RHTAPBUGS-985).
/test appstudio-e2e-tests

**Pull request merged**